lib/Factory/LayoutFactory.php

```diff
@@ -1660,6 +1682,15 @@ public function createNestedPlaylistWidgets($widgets, $combined, &$playlists)
         return $playlists;
     }
 
+    /**
+     * Get all Codes assigned to Layouts
+     * return string[]
+     */
+    public function getLayoutCodes()
```

Comment:
As I've indicated above, my feeling is that this will give us issues if the number of Layouts with codes in a CMS expands above a small handful.

I agree with separating this out into its own DB call (no need to get all the other Layout details), but this ought to implement paging too, so that it can be used in a select2.